internal/datadog/metrics_client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type metricsClient struct {
+	meter  metric.Meter
```

Comment:
To test this, you can use a manual reader and check the resulting metricsdata: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#ManualReader

Reply:
You can also use https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest